lib/browser/api/browser-window.ts

```diff
@@ -44,6 +45,26 @@ BrowserWindow.prototype._init = function (this: BWT) {
     this.on(event as any, visibilityChanged);
   }
 
+  const warn = deprecate.warnOnceMessage('\'scroll-touch-{begin,end,edge}\' are deprecated and will be removed. Please use the WebContents \'input-event\' event instead.');
+  this.webContents.on('input-event', (_, e) => {
+    if (e.type === 'gestureScrollBegin') {
+      if (this.listenerCount('scroll-touch-begin') !== 0) {
+        warn();
+        this.emit('scroll-touch-begin');
+      }
+    } else if (e.type === 'gestureScrollUpdate') {
+      if (this.listenerCount('scroll-touch-edge') !== 0) {
+        warn();
+        this.emit('scroll-touch-edge');
```

Comment:
If I'm reading the old code correctly, wasn't this event fired on `gestureScrollBegin` and `gestureScrollEnd` as well? Is it worth keeping that behavior to the same for now?

Reply:
yeah, you're right, I've added those back.